Add puid to samples and projects

ChrisHuynh333 opened the pull request:
## What does this PR do and why?
This PR adds the PUID to the samples tables (in both projects and groups), as well as a badge on the Project and Sample views containing its PUID.

## Screenshots or screen recordings
Project Samples View:
![image](https://github.com/phac-nml/irida-next/assets/82407232/2801bb65-4d6d-4c26-8e05-65cd80a043c7)

Project Samples View with sorting:
![image](https://github.com/phac-nml/irida-next/assets/82407232/4a40bff3-f949-4e6b-bd49-d5faeeb34783)

Project Samples View with sorting and filtering:
![image](https://github.com/phac-nml/irida-next/assets/82407232/5f21060c-bba0-4c1d-ba4c-0430abcb9e5d)

Group Samples View:
![image](https://github.com/phac-nml/irida-next/assets/82407232/43bc76b6-a279-4830-903c-97b46de8de37)

Group Samples View with sorting:
![image](https://github.com/phac-nml/irida-next/assets/82407232/a745a8a5-6c3a-4b52-ab0e-f483e7323c0a)

Group Samples View with sorting and filtering:
![image](https://github.com/phac-nml/irida-next/assets/82407232/6a1983d0-052f-4586-9187-262551e02b55)

Sample with new puid badge and breadcrumb:
![image](https://github.com/phac-nml/irida-next/assets/82407232/98aaf1c1-1ad2-474d-b5ae-2bc752dc29d5)

Project with new puid badge:
![image](https://github.com/phac-nml/irida-next/assets/82407232/325c3132-0bbe-4440-bb0f-2ac3af2e0b50)

## How to set up and validate locally
1. Navigate to the Projects Samples and Groups Samples views and ensure the PUID for samples are in the table.
2. Ensure you can sort and filter by PUID.
3. Navigate to a sample and project, and ensure you can view the PUID by their respective names.
4. On a sample view, ensure the breadcrumb now contains the accurate PUID, and clicking the PUID breadcrumb links you to the same sample.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

joshsadam commented on app/components/viral/pill_component.rb:
```diff
@@ -13,7 +13,8 @@ class PillComponent < Viral::Component
       purple: 'bg-purple-100 text-purple-800 dark:bg-purple-900 dark:text-purple-300',
-      pink: 'bg-pink-100 text-pink-800 dark:bg-pink-900 dark:text-pink-300'
+      pink: 'bg-pink-100 text-pink-800 dark:bg-pink-900 dark:text-pink-300',
+      primary: 'bg-primary-100 text-primary-800 dark:bg-primary-800 dark:text-primary-400'
```
**praise:** This looks great, thanks

joshsadam commented on app/views/groups/samples/_table.html.erb:
```diff
@@ -41,6 +41,27 @@
           field: :name
         ) %>
       </th>
+      <th
```
**chore:** For all the tables, can you make the ID (puid) the first column this will now become the fixed column for all tables.

ChrisHuynh333 replied:
Fixed in 731734448cf99512fac5eba96ea2260c5183b13e

joshsadam reviewed (changes requested):
This is looking good.  I am going to add a couple things that I would also like to see in this PR:

1. Update the search functionality (from the Filter by name) to now include PUID, please update all placeholder text to include this.
2. Update the breadcrumbs so that when a user ends up on a sample page it now had the PUID (`Yersinia > Yersinia pseudotuberculosis > Outbreak 2023 > Samples > PUID`)

ChrisHuynh333 commented:
> This is looking good. I am going to add a couple things that I would also like to see in this PR:
> 
> 1. Update the search functionality (from the Filter by name) to now include PUID, please update all placeholder text to include this.
> 2. Update the breadcrumbs so that when a user ends up on a sample page it now had the PUID (`Yersinia > Yersinia pseudotuberculosis > Outbreak 2023 > Samples > PUID`)

1. Added in da24717f05bd3549eba643a8bf3d2cf9bc0f5ced
2. Added in 731734448cf99512fac5eba96ea2260c5183b13e

joshsadam commented on app/views/groups/samples/_table.html.erb:
```diff
@@ -90,16 +98,40 @@
+                      },
+                      class: "text-slate-900 dark:text-slate-100 font-semibold hover:underline" do %>
+            <span class="font-semibold">
+              <%= highlight(
```
**praise:** I love this addition (I actually have a tab open with this and was planning on updating the tables with it), thank  you.

joshsadam reviewed (approved):
This looks fantastic.  Thanks for adding the highlighting.
